internal/controller/csiaddons/persistentvolumeclaim_controller.go

```diff
@@ -146,59 +153,65 @@ func (r *PersistentVolumeClaimReconciler) Reconcile(ctx context.Context, req ctr
 	return ctrl.Result{}, keyRotationErr
 }
 
-// checkDriverSupportReclaimsSpace checks if the driver supports space
-// reclamation or not. If the driver does not support space reclamation, it
-// returns false and if the driver supports space reclamation, it returns true.
-// If the driver name is not registered in the connection pool, it returns
-// false and requeues the request.
-func (r *PersistentVolumeClaimReconciler) checkDriverSupportReclaimsSpace(logger *logr.Logger, annotations map[string]string, driver string) (bool, bool) {
-	reclaimSpaceSupportedByDriver := false
-
-	if drivers, ok := annotations[rsCSIAddonsDriverAnnotation]; ok && slices.Contains(strings.Split(drivers, ","), driver) {
-		reclaimSpaceSupportedByDriver = true
-	}
-
-	ok := r.supportsReclaimSpace(driver)
-	if reclaimSpaceSupportedByDriver && !ok {
-		logger.Info("Driver supports spacereclamation but driver is not registered in the connection pool, Reqeueing request", "DriverName", driver)
-		return true, false
-	}
-
-	if !ok {
-		logger.Info("Driver does not support spacereclamation, skip Requeue", "DriverName", driver)
+// checkDriverSupportCapability checks if the given driver supports the specified capability.
+// It is also used to determine the requeue in case the driver supports the capability
+// but the capability is not found in connection pool.
+// The first return value is true when a requeue is needed.
+// The second return value is true when the capability is registered.
+func (r *PersistentVolumeClaimReconciler) checkDriverSupportCapability(
+	logger *logr.Logger,
+	annotations map[string]string,
+	driverName string,
+	cap Operation) (bool, bool) {
+	driverSupportsCap := false
+	capFound := false
+
+	var driverAnnotation string
+	switch cap {
+	case relciamSpaceOp:
+		driverAnnotation = rsCSIAddonsDriverAnnotation
+	case keyRotationOp:
+		driverAnnotation = krCSIAddonsDriverAnnotation
+	default:
+		logger.Info("Unknown capability", "Capability", cap)
```

Comment:
should it be Unknown operation?

Reply:
We check for capability inside this function. The reason constants are called `Operation` and not `Capability` is to avoid ambiguity with CSI's `Capability` struct.